### PR TITLE
Inbox/adhoc fixes

### PR DESCRIPTION
**Pull request by NelsonVides**
Extract from #3462 code that can be considered neutral or plain improvements over the existing code, so that it can be merged to master promptly.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/100416) / small_tests / ee5c39c95929826d2b6c807a5687d59284740c92
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3500/100416/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3500/100416/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-01-13_17.37.39/lib.mongooseim.logs/run.2022-01-13_17.37.40/suite.log.html)

File: big_tests/tests/inbox_helper.erl [hunk 1]
```diff
@@ -506,10 +506,7 @@ leave_room(User, Room, Occupants) ->
     lists:foreach(fun escalus:wait_for_stanza/1, Occupants).
 
 wait_for_groupchat_msg(Users) ->
-    Resps = lists:map(fun(User) -> escalus:wait_for_stanza(User) end,
-              Users),
-    lists:foreach(fun(Resp) -> escalus:assert(is_groupchat_message, Resp) end,
-                  Resps).
+    [ escalus:assert(is_groupchat_message, escalus:wait_for_stanza(User)) || User <- Users].
```
Comment (arcusfelis) [inline big_tests/tests/inbox_helper.erl:509]:
```suggestion
    [ escalus:assert(is_groupchat_message, escalus:wait_for_stanza(User)) || User <- Users].
```
```suggestion
    [escalus:assert(is_groupchat_message, escalus:wait_for_stanza(User)) || User <- Users].
```

File: src/inbox/mod_inbox_rdbms.erl [hunk 2]
```diff
@@ -74,8 +74,9 @@ init(HostType, _Options) ->
                                  UniqueKeyFields),
     rdbms_queries:prepare_upsert(HostType, inbox_upsert_incr_unread, inbox,
                                  InsertFields,
-                                 [<<"content">>, <<"unread_count = inbox.unread_count + 1">>,
-                                  <<"msg_id">>, <<"timestamp">>, <<"archive">>],
+                                 [<<"content">>, <<"msg_id">>, <<"timestamp">>, <<"archive">>,
+                                  {<<"unread_count">>, <<"unread_count = inbox.unread_count + ?">>}
+                                 ],
```
Comment (arcusfelis) [inline src/inbox/mod_inbox_rdbms.erl:79]:
```suggestion
                                  {<<"unread_count">>, <<"unread_count = inbox.unread_count + ?">>}
                                 ],
```
```suggestion
                                  {<<"unread_count">>, <<"unread_count = inbox.unread_count + ?">>}],
```

File: src/inbox/mod_inbox_rdbms.erl [hunk 3]
```diff
@@ -135,15 +136,22 @@ remove_domain(HostType, LServer) ->
 
 %% This function was not refatorected to use the generic upsert helper
```
Comment (arcusfelis) [inline src/inbox/mod_inbox_rdbms.erl:137]: refatorected, becase - typo

File: big_tests/tests/inbox_SUITE.erl [hunk 4]
```diff
@@ -1008,14 +971,14 @@ system_message_is_correctly_avoided(Config) ->
         %% Given a room
         muc_light_helper:given_muc_light_room(Room, Alice, InitOccupants),
         BobRoomJid = <<RoomJid/binary,"/", BobJid/binary>>,
```
Comment (arcusfelis) [inline big_tests/tests/inbox_SUITE.erl:973]: Aaa, we need a code formatter (I know, it is not related to this PR, but could be fixed in this PR maybe)

```suggestion
        BobRoomJid = <<RoomJid/binary,"/", BobJid/binary>>,
```
```suggestion
        BobRoomJid = <<RoomJid/binary, "/", BobJid/binary>>,
```

**Review by arcusfelis — APPROVED** [review]
looks good to me, added just questionable comments :D

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/100436) / small_tests / 15bf752c7ad371a95c5282f586a3fe98fd251976
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3500/100436/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3500/100436/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-01-13_18.38.14/lib.mongooseim.logs/run.2022-01-13_18.38.15/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/100473) / small_tests / 34f38f6454769d08b67cffe2292b8ddffd9a8e44
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3500/100473/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3500/100473/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-01-13_18.59.57/lib.mongooseim.logs/run.2022-01-13_18.59.57/suite.log.html)

**Pull request merged**